### PR TITLE
Py2 take2

### PR DESCRIPTION
**Pull request by sparkiegeek**
@bloodearnest wishes to merge this into master

### CODE REVIEW

File: acceptable/tests/_fixtures.py [hunk 1]
```diff
@@ -8,8 +16,14 @@
 
 from acceptable import _service
 
+PY2 = sys.version_info[0] == 2
```
Comment (sparkiegeek) [inline acceptable/tests/_fixtures.py]: `from future.utils import PY2`

File: acceptable/tests/_fixtures.py [hunk 2]
```diff
@@ -8,8 +16,14 @@
...
+
 
 def clean_up_module(name, old_syspath=None):
+    if PY2:
+        if isinstance(name, str):
```
Comment (sparkiegeek) [inline acceptable/tests/_fixtures.py]: You could squash this up to an `and`

File: acceptable/tests/test_build_doubles.py [hunk 3]
```diff
@@ -12,10 +20,19 @@
...
 
+
 from acceptable import _build_doubles
 
 
-class ExtractSchemasFromSourceTests(TestCase):
+class BuildDoubleTestCase(TestCase):
+
+    def setUp(self):
+        super().setUp()
```
Comment (sparkiegeek) [inline acceptable/tests/test_build_doubles.py]: hmm `super()` will fail on Py2, move it to a decorator a la https://docs.python.org/2/library/unittest.html#skipping-tests-and-expected-failures

File: acceptable/tests/test_main.py [hunk 4]
```diff
@@ -15,12 +25,15 @@
...
 from acceptable import __main__ as main
 from acceptable import get_metadata
-from acceptable.tests.fixtures import (
+from acceptable.tests._fixtures import (
     CleanUpModuleImport,
     TemporaryModuleFixture,
 )
 
 
+PY2 = sys.version_info[0] == 2
```
Comment (sparkiegeek) [inline acceptable/tests/test_main.py]: `future.util` again

File: acceptable/__main__.py [hunk 5]
```diff
@@ -134,23 +143,24 @@ def import_metadata(module_paths):
...
+    modules = []
     try:
         for path in module_paths:
-            import_module(path)
+            modules.append(import_module(path))
     except ImportError as e:
-        raise RuntimeError(
-            'Could not import {}: {}'.format(path, str(e))
-        ) from e
+        err = RuntimeError('Could not import {}: {}'.format(path, str(e)))
```
[the remark this answers is not in the document]
Comment (bloodearnest) [reply acceptable/__main__.py]: Removed

File: acceptable/tests/test_build_doubles.py (hunk 3 above)
Comment (bloodearnest) [reply acceptable/tests/test_build_doubles.py]: argumentless super() works fine via future module

File: acceptable/_service.py [hunk 6]
```diff
@@ -13,7 +19,7 @@ class InvalidAPI(Exception):
     pass
 
 
-class APIMetadata:
+class APIMetadata(object):
```
Comment (cjwatson) [inline acceptable/_service.py]: IME `__metaclass__ = type` at the top is nicer than having to do this everywhere.

File: acceptable/_service.py [hunk 7]
```diff
@@ -113,9 +119,6 @@ def __init__(self, name, group=None, metadata=None):
         :raises TypeError: If the name string is something other than a
             string.
         """
-        if not isinstance(name, str):
-            raise TypeError(
-                "name must be a string, not %s" % type(name).__name__)
```
Comment (cjwatson) [inline acceptable/_service.py]: I think this method's docstring (the `:raises TypeError:` part) is now wrong.

(Alternatively, retain this check but use `past.builtins.basestring`?)

File: acceptable/tests/_fixtures.py [hunk 8]
```diff
@@ -10,6 +17,9 @@
 
 
 def clean_up_module(name, old_syspath=None):
+    if PY2 and isinstance(name, str):
+        name = name.encode('utf8')
```
Comment (cjwatson) [inline acceptable/tests/_fixtures.py]: Maybe `from future.utils import bytes_to_native_str as n` and `sys.modules.pop(n(name))`?

File: acceptable/tests/test_build_doubles.py (hunk 3 above)
Comment (cjwatson) [reply acceptable/tests/test_build_doubles.py]: Yet you seem to have converted away from argumentless `super()` elsewhere?

File: acceptable/tests/test_build_doubles.py [hunk 9]
```diff
@@ -12,10 +18,19 @@
...
+        if sys.version_info[0] == 2:
```
Comment (cjwatson) [inline acceptable/tests/test_build_doubles.py]: Would be nice if these tests were consistent across the PR.  `from future.utils import PY2` and `if PY2:`?

File: setup.py [hunk 10]
```diff
@@ -31,17 +18,22 @@ def parse_requirements_file(path):
...
     license='LGPLv3',
     packages=['acceptable'],
-    install_requires=parse_requirements_file('requirements.txt'),
+    install_requires=[
+        'Flask<1.0',
+        'jsonschema',
+        'pyyaml',
+        'jinja2',
+        'future',
+    ],
```
Comment (cjwatson) [inline setup.py]: Sort these while you're here?

**Pull request merged**